### PR TITLE
refactor: mqtt permissions check

### PR DESCRIPTION
**Pull request by caspiano**
Use a flag enum to represent the access control flags provided by the [mosquito go auth client](https://github.com/iegomez/mosquitto-go-auth/blob/master/backends/constants/constants.go)

### CODE REVIEW

File: src/placeos-rest-api/controllers/root.cr [hunk 1]
```diff
@@ -102,40 +120,63 @@ module PlaceOS::Api
...
-      mqtt_parse_token
       head :ok
     end
 
+    @[Flags]
+    enum MqttAcl
+      Read      = 0x01
+      Write     = 0x02
+      Subscribe = 0x04
+      Deny      = 0x11
```
Comment (stakach) [inline src/placeos-rest-api/controllers/root.cr:131]: Not sure this needs to be a flags, not sure where deny came from either?

The values mosquitto pass are:
1 is read, 2 is write, 3 is readwrite, 4 is subscribe

**Review by stakach — CHANGES_REQUESTED** [review]

File: src/placeos-rest-api/controllers/root.cr (hunk 1 above)
Comment (caspiano) [reply src/placeos-rest-api/controllers/root.cr:132]: Yep, they're flags. https://github.com/iegomez/mosquitto-go-auth/blob/master/backends/constants/constants.go

Comment (caspiano) [reply src/placeos-rest-api/controllers/root.cr:132]: Read | Write => 0x01 | 0x02 => 0x03

Comment (stakach) [reply src/placeos-rest-api/controllers/root.cr:132]: makes sense, although probably worth adding a comment explaining the logic there.
man, you went deep

Comment (stakach) [reply src/placeos-rest-api/controllers/root.cr:132]: and add that link to the source code in the comment too

**Review by stakach — CHANGES_REQUESTED** [review]

**Pull request merged**